File: Data Structures & Algorithms/implement-stack-using-queues/submission-1.java
class MyStack {
    private Queue<Integer> q0;
    private Queue<Integer> q1;
    int currentQueue = 0;

    public MyStack() {
        q0 = new LinkedList<>();
        q1 = new LinkedList<>();
    }
    
    public void push(int x) {
        Queue<Integer> q = (currentQueue == 0) ? q0 : q1;
        q.add(x);
    }
    
    public int pop() {
        Queue<Integer> q = (currentQueue == 0) ? q0 : q1;
        Queue<Integer> other = (currentQueue != 0) ? q0 : q1;

        while (q.size() > 1) {
            other.add(q.remove()); // Enqueue all but last
        }
        currentQueue = (currentQueue + 1) % 2;
        return q.remove();
    }
    
    public int top() {
        Queue<Integer> q = (currentQueue == 0) ? q0 : q1;
        Queue<Integer> other = (currentQueue != 0) ? q0 : q1;

        while (q.size() > 1) {
            other.add(q.remove()); // Enqueue all but last
        }

        int value = q.remove();
        currentQueue = (currentQueue + 1) % 2;
        other.add(value);
        return value;
    }
    
    public boolean empty() {
        return q0.isEmpty() && q1.isEmpty();
    }
}

/**
 * Your MyStack object will be instantiated and called as such:
 * MyStack obj = new MyStack();
 * obj.push(x);
 * int param_2 = obj.pop();
 * int param_3 = obj.top();
 * boolean param_4 = obj.empty();
 */

File: Data Structures & Algorithms/design-hashset/submission-1.py
class Node:
    def __init__(self, val):
        self.val = val
        self.next = None

class MyHashSet:

    def __init__(self):
        self.hash_set = [Node(0) for _ in range(10**4)]

    def add(self, key: int) -> None:
        cur = self.hash_set[key % len(self.hash_set)]
        while cur.next:
            if cur.next.val == key:
                return
            cur = cur.next
        cur.next = Node(key)

    def remove(self, key: int) -> None:
        cur = self.hash_set[key % len(self.hash_set)]
        while cur.next:
            if cur.next.val == key:
                cur.next = cur.next.next
                return
            cur = cur.next

    def contains(self, key: int) -> bool:
        cur = self.hash_set[key % len(self.hash_set)]
        while cur.next:
            cur = cur.next
            if cur.val == key:
                return True
        return False



# Your MyHashSet object will be instantiated and called as such:
# obj = MyHashSet()
# obj.add(key)
# obj.remove(key)
# param_3 = obj.contains(key)